tests/Unit/PackagePrivate/Humanizer/Internationalization/ArrayMessageBuilderTest.php

```diff
@@ -50,30 +47,71 @@ public function testParameters(): void {
 			$builder->buildMessage( 'right-key', '2021-02-21', 'Jeroen De Dauw' )
 		);
 
+	}
+
+	public function testPluralArgumentIsSingular(): void {
+		$builder = new ArrayMessageBuilder(
+			[
+				"edtf-day-and-year" => "$1e{{PLURAL:$1|r|}} jour d’un mois inconnu de $2"
+			]
+		);
+
 		$this->assertSame(
 			'1er jour d’un mois inconnu de 1985',
 			$builder->buildMessage( 'edtf-day-and-year', '1', '1985' )
 		);
+	}
+
+	public function testPluralArgumentIsPlural(): void {
+		$builder = new ArrayMessageBuilder(
+			[
+				"edtf-day-and-year" => "$1e{{PLURAL:$1|r|}} jour d’un mois inconnu de $2"
+			]
+		);
 
 		$this->assertSame(
 			'3e jour d’un mois inconnu de 1985',
 			$builder->buildMessage( 'edtf-day-and-year', '3', '1985' )
 		);
+	}
+
+	public function testPluralMultipleArguments(): void {
+		$builder = new ArrayMessageBuilder(
+			[
+				"multiple-arguments" => "day{{PLURAL:$1||s}} $1 and $2 month{{PLURAL:$2||s}}",
+			]
+		);
 
 		$this->assertSame(
-			'... and 1 month',
-			$builder->buildMessage( 'month-number', '1' )
+			'day 1 and 3 months',
+			$builder->buildMessage( 'multiple-arguments', '1', '3' )
+		);
+	}
+
+	public function testPluralMultipleArgumentsInverted(): void {
+		$builder = new ArrayMessageBuilder(
+			[
+				"multiple-arguments-inverted" => "day{{PLURAL:$1||s}} $1 and $2 month{{PLURAL:$2||s}}",
+			]
 		);
 
 		$this->assertSame(
-			'... and 3 months',
-			$builder->buildMessage( 'month-number', '3' )
+			'days 3 and 1 month',
+			$builder->buildMessage( 'multiple-arguments-inverted', '3', '1' )
+		);
+	}
+
+	public function testPluralMultipleArgumentsMissingParameter(): void {
+		$builder = new ArrayMessageBuilder(
+			[
+				"multiple-arguments-missing-parameter" => "day{{PLURAL:$1||s}} $1 and $2 month{{PLURAL:$2||s}}",
```

Comment:
It is nice to keep tests are simple as possible for the case they test. This input is simpler: `a {{PLURAL:$1|b|c}} d`.  Also makes it clear what actually happens: either you get nothing, or you get the first argument. I can't tell that from the current test.